getVfRep.py

```diff
@@ -0,0 +1,98 @@
+        logger.info(f"Generated Server Pod Yaml {self.out_file_yaml}")
+        
+
+    def extract_vf_rep(self) -> str:
+        get_vf_rep_cmd = f'exec -n default {self.pod_name} -- /bin/sh -c "crictl --runtime-endpoint=/host/run/crio/crio.sock ps -a --name={self.iperf_pod_name} -o json "'
+        vf_rep_json = self.run_oc(get_vf_rep_cmd).out
+        data = json.loads(vf_rep_json)
+        return data["containers"][0]["podSandboxId"][:15]
+
+    def run_ethtool_cmd(self, vf_rep: str) -> (int, int):
+        self.ethtool_cmd = f'exec -n default {self.pod_name} -- /bin/sh -c "ethtool -S {vf_rep}"'
+        ethtool_output = self.run_oc(self.ethtool_cmd).out
+        #print("DEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEBAAAAAAAAAAAAAG %s" % ethtool_output)
+        rxpacket = self.parse_out_packet(ethtool_output, "rx_packet")
+        txpacket = self.parse_out_packet(ethtool_output, "tx_packet")
+        return (rxpacket, txpacket)
+
+    def parse_out_packet(self, output: str, prefix: str) -> Optional[int]:
+        for line in output.splitlines():
+            stripped_line = line.strip()
+            if stripped_line.startswith(prefix):
+                return int(stripped_line.split(":")[1])
+
+        return None
+    
+    def run_st(self) -> Result:
+        vf_rep = self.extract_vf_rep()
+
+        (rxpacket_start, txpacket_start) = self.run_ethtool_cmd(vf_rep)
+        (rxpacket_end, txpacket_end) = self.run_ethtool_cmd(vf_rep)
+
+        return (rxpacket_start, txpacket_start, rxpacket_end, txpacket_end)
+
+    def run(self, duration: int):
+        self.exec_thread = ReturnValueThread(target=self.run_st)
+        self.exec_thread.start()
+
+    def stop(self):
+        logger.info(f"Stopping Get Vf Rep execution on {self.pod_name}")
+        r = self.exec_thread.join()
+        #if r.returncode != 0:
+            #logger.info(r)
+
+        #logger.debug(f"GetVfRep.stop(): {r.out}")
+        #logger.debug(f"GetVfRep.stop().vf_rep: {self.vf_rep}")
+
+        data = {"rx_packet_start": r[0], "tx_packet_start": r[1], "rx_packet_end": r[2], "tx_packet_end": r[3]}
+        self._output_ethtool = self.generate_output_ethtool(data, self.ethtool_cmd)
+    
+    def output(self, out: TftAggregateOutput):
+        out.plugins.append(self._output_ethtool) 
+
+        # Print summary to console logs
+        rx_packet_start = self._output_ethtool.result["rx_packet_start"]
+        tx_packet_start = self._output_ethtool.result["tx_packet_start"]
+        rx_packet_end = self._output_ethtool.result["rx_packet_end"]
+        tx_packet_end = self._output_ethtool.result["tx_packet_end"]
+
+        logger.info("rx_packet_start: %d\n tx_packet_start: %d\n rx_packet_end: %d\n tx_packet_end: %d\n" % (rx_packet_start,tx_packet_start,rx_packet_end,tx_packet_end))
+
+    def generate_output_ethtool(self, data, cmd: str) -> PluginOutput:
+        return PluginOutput(
+            plugin_metadata={
+                "name": "GetEthtoolStats",
```

Comment:
This could also be a bit more specific. We should use the same plugin name everywhere i.e. pod name, plugin name provided by the user, class name etc. Similar to measureCpu/Power

Reply:
I am planning on creating at least 2 plugin outputs for this class (GetEthtoolStats as well as TCP dump stats). This class is really going to encapsulate all of HWOL and generate more than 1 plugin output so we can't have naming consistency.